AnnService/src/Core/KDT/KDTIndex.cpp

```diff
@@ -44,6 +44,22 @@ namespace SPTAG
             return ErrorCode::Success;
         }
 
+        template <typename T>
+        ErrorCode Index<T>::LoadIndexData(const std::vector<std::istream*>& p_indexStreams)
+        {
+            if (!m_pSamples.Load(*p_indexStreams[0])) return ErrorCode::Fail;
+            if (!m_pTrees.LoadTrees(*p_indexStreams[1])) return ErrorCode::Fail;
+            if (!m_pGraph.LoadGraph(*p_indexStreams[2])) return ErrorCode::Fail;
+            if (!m_deletedID.Load(*p_indexStreams[3])) return ErrorCode::Fail;
+
+            omp_set_num_threads(m_iNumberOfThreads);
+            m_workSpacePool.reset(new COMMON::WorkSpacePool(max(m_iMaxCheck, m_pGraph.m_iMaxCheckForRefineGraph), GetNumSamples()));
+            m_workSpacePool->Init(m_iNumberOfThreads);
+            m_threadPool.init();
```

Comment:
can init() fail?

Reply:
Init will only fail due to memory overflow. Let us leave it for later commits to add more memory allocation checks.